src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -139,6 +140,7 @@ static ProjectCodeRefactoringProvider.ExtensionInfo GetExtensionInfo(ExportCodeR
                             RefactoringToMetadataMap.TryGetValue(provider, out var providerMetadata);
 
                             using (addOperationScope(providerName))
+                            using (Logger.LogBlock(FunctionId.Refactoring_CodeRefactoringService_GetRefactoringsAsync, KeyValueLogMessage.Create(LogType.UserAction, m => CreateLogProperties(m, provider)), cancellationToken))
```

Comment:
what's the difference between this and the `RoslynEventSource.LogInformationBlock` below?

Reply:
RoslynEventSource is used to send ETW events (which is used as perf measurements), and this one is used to send telemetry.

Reply:
To be more specific, I think ETW is more likey to be used in perf analysis in perf view and speedometer tests.
And here (even we are use the telemetry to measure perf), it is loaded into our db for data analysis purposes

Reply:
I though Logger.LogBlock would also send ETW events though? I'll have to look at the provider though

Reply:
Ah yeah, we have several loggers aggregated toghtgher https://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/Telemetry/VisualStudioWorkspaceTelemetryService.cs,98d90149b4e18754,references

Reply:
Will update this part

Reply:
I change the event name here, (because previously we use the same event for both single provider and all providers, which seems wrong)
And the ETW I just leave the code here since I see a lot of other similar usages. (Logger + RoslynEventSource)
We may end up with logging two ETW events. And we might address this issue in a seperate PR